pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -854,6 +855,29 @@ protected BookieAffinityGroupData internalGetBookieAffinityGroup() {
         }
     }
 
+    public void setNamespaceBundleAffinity (String bundleRange, String destinationBroker) {
+        if (destinationBroker != null) {
+            if (!this.isLeaderBroker()) {
+                LeaderBroker leaderBroker = pulsar().getLeaderElectionService().getCurrentLeader().get();
+                String leaderBrokerUrl = leaderBroker.getServiceUrl();
+                try {
+                    URL redirectUrl = new URL(leaderBrokerUrl);
+                    URI redirect = UriBuilder.fromUri(uri.getRequestUri()).host(redirectUrl.getHost())
```

Comment:
also, it's better to separate out this logic in a separate method
eg: `validateLeader(httpRequest)` this sends redirect response by throwing temporaryRedirect-Exception if current broker is not leader.

Reply:
addressed